pyproject.toml

```diff
@@ -91,8 +91,7 @@ upload_to_release = true
 remove_dist = false
 # Setting build_command to false here b/c `poetry build -vvv` will be used prior to a PSR publish command
 build_command = false
-# Update the script options documentation automatically for each release
-pre_commit_command = "poetry run rich-codex --verbose --skip-git-checks --no-confirm"
+# The `pre_commit_command` option has proven troublesome so these files get updated in the release workflow instead
```

Comment:
The short answer is that it is not _absolutely_ required, but keeping it as a dev dependency is still desired.

The original justifications for adding `rich-codex` as a dev dependency, which is really only a dependency needed for release, were:

1. It allows for keeping tabs on the `rich-codex` project via Phylum analysis of the lockfile. This project is relatively new and is churning through new releases. Pinning the version with the `poetry.lock` file helps to limit surprises when it comes time to release.
2. It ensures that the pre-commit command run as part of the `semantic-release` will be available, regardless of how/where the `semantic-release` is initiated.

Reason 2. is no longer in play...which is a bit of a shame. Reason 1. still holds and should be enough to sway the decision to keep it.

If `rich-codex` is removed as a dev dependency, then it will need to be installed directly during the release workflow instead. That is certainly possible but it would mean either manually pinning the version there and having to remember to update it (e.g., `pipx install rich-codex==1.2.4`) or not pinning it and risk breaking changes (e.g., `pipx install rich-codex`).

One thing that should help in the future is to use optional groups when Poetry 1.2 is available, using [the groups feature](https://github.com/python-poetry/poetry/issues/1644). That would allow for this and other dependencies to go into something like a `ci` group that only gets installed in the CI environment.